service/src/test/java/com/epam/rft/atsy/service/impl/PasswordChangeServiceImplTest.java

```diff
@@ -34,144 +45,280 @@
     private UserRepository userRepository;
 
     @InjectMocks
-    private PasswordChangeServiceImpl sut;
+    private PasswordChangeServiceImpl passwordChangeService;
+
+    private PasswordHistoryDTO passwordHistoryDtoWithUserId;
+    private PasswordHistoryEntity passwordHistoryEntityWithUserEntity;
+    private UserEntity userEntityWithId;
+    public static final String PASSWORD_PREFIX = "password";
+
+    @Before
+    public void setUp() {
+        passwordHistoryDtoWithUserId = PasswordHistoryDTO.builder()
+                .userId(USER_ID)
+                .build();
+
+        passwordHistoryEntityWithUserEntity = PasswordHistoryEntity.builder()
+                .userEntity(userEntityWithId)
+                .build();
+
+        userEntityWithId = UserEntity.builder()
+                .id(USER_ID)
+                .build();
+
+        given(this.modelMapper.map(passwordHistoryDtoWithUserId, PasswordHistoryEntity.class)).willReturn(passwordHistoryEntityWithUserEntity);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void saveOrUpdateShouldThrowIAEWhenPasswordHistoryDtoIsNull() {
+        //Given
+
+        //When
+        this.passwordChangeService.saveOrUpdate(null);
+
+        //Then
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void saveOrUpdateShouldThrowIAEWhenPasswordHistoryUserIdIsNull() {
+        //Given
+        PasswordHistoryDTO passwordHistoryDTO = PasswordHistoryDTO.builder()
+                .userId(null)
+                .build();
+
+        //When
+        this.passwordChangeService.saveOrUpdate(passwordHistoryDTO);
+
+        //Then
+    }
+
+    @Test(expected = DataAccessException.class)
+    public void saveOrUpdateDelegateDataAccessExceptionFromUserRepository() {
+        //Given
+        given(this.userRepository.findOne(USER_ID)).willThrow(DataIntegrityViolationException.class);
+
+        //When
+        this.passwordChangeService.saveOrUpdate(passwordHistoryDtoWithUserId);
+
+        //Then
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void saveOrUpdateShouldThrowIAEWhenUserCannotBeFoundById() {
+        // Given
+        given(this.userRepository.findOne(USER_ID)).willReturn(NULL_USER_ENTITY);
+        given(this.passwordHistoryRepository.findByUserEntity(NULL_USER_ENTITY)).willThrow(IllegalArgumentException.class);
```

Comment:
This given statement is not necessary as the Assert.notNull(userEntity) will throw the IAE before the control gets to passwordHistoryRepository